pkg/version/version.go

```diff
@@ -49,3 +51,36 @@ func Get() Info {
 		Platform:   fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
 	}
 }
+
+// Error is an error for versions in case it is desired to check an error for this type
+type Error struct {
+	Component       string
+	ExpectedVersion string
+	Version         string
+}
+
+// Error is a required function for type Error
+func (e Error) Error() string {
+	return fmt.Sprintf("expected version: %s, found vresion: %s", e.ExpectedVersion, e.Version)
+}
+
+//Valid returns nil if expected version is meet by actual using solely Major.Minor
+func Valid(component string, actual *semver.Version, expected *semver.Version) error {
+	if actual.Major() < expected.Major() || actual.Minor() < expected.Minor() {
+		return Error{
+			Component:       component,
```

Comment:
what is a component? 🤔 it's passed in only for logging?

Reply:
what is the advantage of doing this VS returning bool or -1, 0, 1 and handling error one level up where all the context for logging including component is already in place?

Reply:
the semver.Compare doesn't work for us as we don't want to have patch as part of compare.  To write our own generalized compare doesn't seem like the right way to go as it will add more code with no additional value.   

Regarding component... yeah... I wanted to log detailed info at the point we know the most... that said, it would be easy to wrap the version error, but it puts a burden on the next caller of this function to know they need to do the same, where this encourages the right expectations.

Reply:
well this is not really validating anything to me, it's really comparing. So maybe CompareMajorMinor?

please at least document what is the component, as a called I would have no idea what to pass in...

Reply:
The reason this was not a compare -1, 0, 1 or a bool is because I wanted to provide the reason for the result... in the end I'm just printing the full sem ver... this can be pulled out with the calling providing the error (which removes the component issue as well).   un momento, por favor